services/QuillLMS/app/services/stripe_integration/webhooks/invoice_paid_event_handler.rb

```diff
@@ -3,15 +3,22 @@
 module StripeIntegration
   module Webhooks
     class InvoicePaidEventHandler < EventHandler
+      class ZeroAmountPaidError < StandardError; end
+
       PUSHER_EVENT = 'stripe-subscription-created'
 
       def run
+        check_for_zero_paid
         create_subscription unless manual_invoice?
         stripe_webhook_event.processed!
       rescue => e
         stripe_webhook_event.log_error(e)
       end
 
+      private def check_for_zero_paid
```

Comment:
You might be able to check the `subscription.status` == 'active' to confirm this: https://stripe.com/docs/billing/subscriptions/webhooks#events

![Screen Shot 2022-06-16 at 9 43 16 AM](https://user-images.githubusercontent.com/1304933/174083546-213e00ed-d2ab-4c0b-abce-d61f995bb823.png)

Reply:
Good idea.  I will investigate.